File: 01/Wanzaz.py
# 1. Solution wih a simple code
number = int(input("Enter a number: "))
for num in range(1,number):
    string = ""
    if num % 3 == 0:
        string = string + "Fizz"
    if num % 5 == 0:
        string = string + "Buzz"
    if num % 5 != 0 and num % 3 != 0:
        string = string + str(num)
    print(string)


# 2. Solution with Lambda function
# print(*map(lambda i: 'Fizz'*(not i%3)+'Buzz'*(not i%5) or i, range(1, number)),sep='\n')

File: 01/Wanzaz.rs
use std::io::{self, Write};

fn main() {
    print!("Enter a integer number: ");
    io::stdout().flush().unwrap();

    let mut input_text = String::new();
    io::stdin()
        .read_line(&mut input_text)
        .expect("failed to read from stdin");

    let trimmed = input_text.trim();
    match trimmed.parse::<i32>() {
        Ok(i) => {
            for index in 1..i {
                if index % 3 == 0 && index % 5 == 0 {
                    println!("FizzBuzz")
                } else if index % 3 == 0 {
                    println!("Fizz")
                } else if index % 5 == 0 {
                    println!("Buzz")
                } else {
                    println!("{}", index)
                }
            }
        }        
        Err(..) => println!("this was not an integer: {}", trimmed),
    };
}
